### DISCUSSION TITLE
Analysis of Battaglia Profile and sim results

### DISCUSSION THREAD

**Discussion by kbanker** [deepskies/DeepSZSim#123] [General]
Continuing from issue #93 to discuss the results of the sims, both in isolation and compared to the sims from Sehgal 2010.

**Comment by bnord** [comment]
is this latest plot a difference plot? 
https://github.com/deepskies/SimSZ/issues/93#issuecomment-1664141990

**Reply by kbanker** [reply]
Yes, the last plot there is an average over 1000 loops of difference plots (Seghal sim - DeepSZSim).

**Reply by bnord** [reply]
Could you make a version where the z axis is a fractional error or a percentage? That will help us do some quick evaluations. 

Could we also do a little looping over redshift ranges and mass ranges and make multiple of these plots? 

e.g., 

```
for iz in zwindows:
    for im in mwindows:
        make_residual_tsz_map(iz, im)
```

**Comment by kbanker** [comment]
Here are the average difference maps for a couple different mass bins for 0<z<0.3 redshift
<img width="532" alt="Screen Shot 2023-08-09 at 4 49 18 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/d53500a3-8644-48d2-b041-7c9d0c7e5279">
<img width="530" alt="Screen Shot 2023-08-09 at 4 29 44 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/9884f57b-4c8f-4d4e-994e-edb35f5f95a9">
<img width="545" alt="Screen Shot 2023-08-09 at 5 22 45 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/334897b9-e15a-4b4f-bb49-3a2da7b79f45">

**Comment by kbanker** [comment]
And these are for 1.1e14 < M200 < 1.7e14; but varying redshift (The first map in the same as the one in the previous comment
<img width="530" alt="Screen Shot 2023-08-09 at 4 29 44 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/2a8b0086-eb94-4109-90cc-e970c2257c2b">
<img width="554" alt="Screen Shot 2023-08-09 at 5 38 38 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/d518f96f-57dc-46f1-99b0-f80102664097">
<img width="559" alt="Screen Shot 2023-08-09 at 5 53 12 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/be691447-1185-442a-b0d2-bf0eefe892b2">

**Reply by bnord** [reply]
The highest-mass bin having the smoothest map makes sense to me, because I expect the most massive structure to have smooth profiles or signals that are much higher than the noise.

**Reply by bnord** [reply]
I'm not sure how to interpret the high-redshift smoothness.

**Comment by kbanker** [comment]
After doing a percent difference graph (Seghal - DeepSZ) /DeepSZ, Im getting really high values around the edge, probably because our sim has really low values at those points. Here it is in log-scale:
<img width="551" alt="Screen Shot 2023-08-09 at 8 05 36 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/d2d30021-43c6-47b0-a7f9-aafaf7c864f9">

**Reply by bnord** [reply]
It seems to be very high in the center also. it would be $10^3\%$ at the center. I think something might be going wrong here? Are you dividing by $1000$?

**Reply by kbanker** [reply]
That was actually the problem, I thought that I was dividing by 1000 since I had percent_map /num_loops, but I forgot I redefined num_loops for some other test runs, so it wasn't dividing by 1000. Here's the map with the correct division, that should represent percent error (so it includes the * 100):
<img width="554" alt="Screen Shot 2023-08-10 at 5 55 47 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/b3249208-869e-4722-9b94-b37ed2071f91">

**Reply by kbanker** [reply]
Here the central pixel has 2.8% error, but goes up to 31392% at max.

**Reply by bnord** [reply]
@kbanker Could you add contours with labels so we can see more readily where the percent error?

**Reply by bnord** [reply]
@kbanker will you still make the rows or grids of these?

**Reply by kbanker** [reply]
The grid with uniform color bars is definitely something I am going to do. I am more unsure on how exactly I would add counters with labels but I can look into it (probably something in Matplotlib)

**Reply by bnord** [reply]
If there's something easy in matplotlib lets you add contours, I think it would be good to do, but if it's not easy/quick, it's probably not worth it.

**Reply by bnord** [reply]
you might have to use a 2D histogram function, I'm not sure.

**Comment by kbanker** [comment]
Here are the 4 mass bins in plots with absolute difference in uK being on the color bars. Here they are all on the same color scale so maybe a bit more comparable than before.
<img width="563" alt="Screen Shot 2023-08-12 at 3 08 23 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/2e5f9f97-9fa0-4667-bb4e-47f01597a545">

**Reply by bnord** [reply]
1. is it possible to still do percentages for the z-axis?
1. Could you make the title font smaller so it's a little easier to read? alternatively, could you put the mass and redshift range within each little map instead of the title?

**Reply by kbanker** [reply]
I still need to compute all the percentage maps, since I was adding contours, but here's the one that I have (just added contours). Will try to reduce the size of the title when I generate this for the percent maps.
<img width="555" alt="Screen Shot 2023-08-12 at 8 58 30 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/bdc42f5e-7805-4fd6-9d02-0f565aa08da3">

**Reply by bnord** [reply]
interesting. These errors seem quite large. @evavagiakis What do you think about the errors here?

**Reply by evavagiakis** [reply]
I'd encourage talking about these as "differences" rather than errors, as we aren't expecting to reproduce the substructures and offcentering present in the independent hydrosim set. I do think the percentage differences are rather large and don't seem to jive with my understanding of the differences when presented in uK

**Comment by kbanker** [comment]
Here is the grid with the percentage maps, and clearer title size, along with contours. 
<img width="557" alt="Screen Shot 2023-08-14 at 3 10 03 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/3849847b-fd20-4ba2-8cc5-efe49eb9332c">

**Reply by bnord** [reply]
What are the x-axis and y-axis values? pixels? arcmin?
These percentages seem really high.

**Reply by kbanker** [reply]
Yeah, sorry for not adding all the axes titles, but x and y are in arcmin and the values are still in percent error.

I agree that's they're really high, but I do think that is because I'm dividing by our profile, which is extremely small when away from the center (so small absolute errors lead to large percent errors)

**Reply by evavagiakis** [reply]
What's the entire formula you're using to determine the plotted percentage difference?

**Reply by kbanker** [reply]
Just while thinking about this, I tried running one of these the other way (so instead of (seghal -deepsz)/deepsz I did (signal -deepsz)/seghal), and this is the result I got, with far lower values around the edges.
<img width="548" alt="Screen Shot 2023-08-14 at 4 48 11 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/1d8027df-9ea2-474a-9a6b-4c227dd5dac1">

**Reply by kbanker** [reply]
> What's the entire formula you're using to determine the plotted percentage difference?

In the grid above I was using AVERAGE { (Seghal - DeepSZ)/DeepSZ } * 100%

**Reply by bnord** [reply]
In case you're curious, You can also do basic latex in github

`$\langle \frac{sehgal - deepsz}{deepsz} \rangle * 100$`
-->
$\langle \frac{sehgal - deepsz}{deepsz} \rangle * 100$

**Comment by kbanker** [comment]
Here are the (Seghal - Deepsz)/Seghal maps in a 3x3 grid; first one is percent difference, then next one is absolute difference.

<img width="864" alt="Screen Shot 2023-08-16 at 5 56 04 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/be81d44e-043f-4815-8dcb-c8b800c85ac5">

<img width="856" alt="Screen Shot 2023-08-16 at 5 59 33 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/58a35926-4ae5-4c2f-8112-8f644b644973">

**Reply by bnord** [reply]
these are pure tSZ with now noise or beam. is that correct?

**Reply by kbanker** [reply]
Yeah exactly, I'm now regenerating these zoomed in, then I'll do it with beam convolution

**Comment by kbanker** [comment]
Here are the zoomed in versions of the pervious maps (all from -4 to 4 arcmin) same bins as above. First map is percent difference, second one is absolute difference in uK
<img width="858" alt="Screen Shot 2023-08-17 at 5 14 18 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/ca746aec-d700-453b-9448-db5ebdabe30c">
<img width="847" alt="Screen Shot 2023-08-17 at 5 16 53 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/c83a86de-efb9-4779-82a1-8c039de81ba7">

**Comment by kbanker** [comment]
Same as above but with a beam convolution added on both maps before the difference was taken. You can definitely see the edge effects from the convolution on the percent difference map. First map in percent difference, second is absolute difference in uK
<img width="878" alt="Screen Shot 2023-08-17 at 10 02 47 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/6fda90e9-f92c-4859-a4bf-aa977e0d26ed">
<img width="842" alt="Screen Shot 2023-08-17 at 10 03 38 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/9f8f63b5-5517-4c28-b673-694f3e61dc30">

**Reply by kbanker** [reply]
I will probably work on generating 1-d profiles next from these maps.

**Reply by evavagiakis** [reply]
That sounds great, it will be nice to see the various mass and redshift trends all in one 1-d profile plot. I think we need to be cropping the edges when we beam-convolve in general

**Reply by bnord** [reply]
`cropping the edges when we beam-convolve in general`
 cropping the overall image? or within a contour?

**Reply by bnord** [reply]
It looks like the map differences are at the $\sim 5$% level between raw and convolved?

**Reply by evavagiakis** [reply]
After beam convolving there is a ~1 pixel edge effect that I believe we currently remove after beam convolution because it presents a "frame" around the image that is an artifact/not useful

**Comment by kbanker** [comment]
Here are the 1-d profiles varying over z (calculated using the zoomed in percent difference maps without beam convolution)
<img width="619" alt="Screen Shot 2023-08-21 at 3 49 19 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/51498b51-9c82-433e-9148-6a45b02f37c6">

**Reply by bnord** [reply]
Coud you update the colors to be colorblind-friendly?

**Reply by kbanker** [reply]
Here are the profiles for varying mass, redshift between 0-0.3:
<img width="588" alt="Screen Shot 2023-08-21 at 3 58 36 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/88de4893-1463-472b-83b5-896c769424d7">

**Reply by bnord** [reply]
is this a slice or a sum around the azimuth?

**Reply by kbanker** [reply]
I can definitely update the colors, will find a colorblind-friendly palette.

**Reply by kbanker** [reply]
These are done by radially averaging around the azimuth

**Reply by bnord** [reply]
I guess taking the average in a number of annuli?

**Reply by bnord** [reply]
could you also put these two together into a 2x2 grid, where the top row is no beam and the bottom has beam included?

**Reply by kbanker** [reply]
Here they are with colorblind friendly colors:
<img width="587" alt="Screen Shot 2023-08-21 at 4 05 41 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/850d5982-ec61-4954-ac9f-abfffa06b593">
<img width="626" alt="Screen Shot 2023-08-21 at 4 06 19 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/89620699-64d6-4797-b857-51f235bf4d32">

**Reply by kbanker** [reply]
Actually I didn't specify annuli, just took each point along with it's radial distance from the origin, and then averaged the points that shared a radial distance.

**Reply by kbanker** [reply]
I can definitely do these for the beam-convolved versions, but I worry that the edge effect "frame" will cause some weird effects

**Reply by bnord** [reply]
I think that is pretty close to doing annuli

**Reply by bnord** [reply]
I htink the edge effect only occurs along the edges of the image. Is that correct? If so, I think we can just pay attention to that at high radii.

**Reply by kbanker** [reply]
Yeah, the effect only occurs at the edge, but this is what we get with the convolution (Maybe it makes sense to chop it off before the edge effects start?)
<img width="616" alt="Screen Shot 2023-08-21 at 4 23 06 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/b022e7c5-cdd8-45f8-97fb-aa59f7333b1a">

**Reply by bnord** [reply]
we can keep it the edges for now. It's good information to have.

**Comment by kbanker** [comment]
Here's the 2x2 grid, varying over mass and redshift. Lower row is beam convolved. As in the single example above, the redshift for the mass-varying profiles is 0-0.3, and for the redshift varying profiles we have 1.1e14 SM < M200 < 1.7e14 SM 
<img width="884" alt="Screen Shot 2023-08-21 at 11 36 43 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/1e771820-dc78-4f2c-92e4-fc05fdc48ada">

**Reply by bnord** [reply]
interesting. 

Here are my observations
1. the difference in the profile shape changes by redshift
2. the magnitude of the difference in the profile changes by mass
3. the convolution causes a slight change in the shape

@kbanker I think you were right in the context of comparisons: we should make another version that disregards everything after ~3.5 arcmin. This will help us view more clearly visually the difference between non-convolved and convolved. 

@kbanker Could you make one more version of this 2x2 in that manner?

**Reply by evavagiakis** [reply]
- It makes sense to me that the higher mass Battaglia fits are in better agreement with the sims, because in general the higher-mass halos are better behaved in terms of profile fits (higher signal and more consistent shape). 
- My interpretation of the redshift trend is that lower redshift objects are more extended in arcmin and therefore reduce the differences between the Battaglia fits and the extended structure present in the Seghal sims over a larger area. 
- It looks like the convolution reduces the differences which would also make sense to me due to blurring out some of that Seghal structure detail we don't represent in our Battaglia fits, but agree it would be nice to see these plots without the poorly behaved region at the edges

**Reply by kbanker** [reply]
Here are the same plots but all with the same scaling and with a cutoff at 3.5 arcmin
<img width="891" alt="Screen Shot 2023-08-22 at 7 16 00 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/9e201dc1-2754-406d-925a-cba04b1cb12e">

**Reply by bnord** [reply]
I'm wondering if we should put the convolved curves in the same plot as the non-convolved curves but with a different line shape (like dashed?)

**Reply by kbanker** [reply]
Here it is, beam convolved are dashed. Looks like there is not much difference at all between beam convolved vs not beam convolved (or all the differences were averaged out during the azimuthal average)
<img width="913" alt="Screen Shot 2023-08-23 at 2 27 14 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/059849f1-2169-4394-9db5-553d1fe30a79">

**Reply by bnord** [reply]
this is great. 

@evavagiakis Do you see anything here that indicates a problem? Is this this time to share with Camille for her feedback?

**Reply by evavagiakis** [reply]
I'd add the dashed line definition to the legend and the fixed mass (for varying z) and redshift (for varying M) value to the subplot titles, otherwise looks great and we should ping Camille

**Comment by kbanker** [comment]
@evavagiakis @bnord Here is the final plot as we talked about
<img width="885" alt="Screen Shot 2023-08-25 at 4 22 58 PM" src="https://github.com/deepskies/SimSZ/assets/56738307/1421a4d5-1a42-4d01-8467-485a837ebf38">

**Reply by bnord** [reply]
@kbanker Could you write out 50-150-word explanation of this plot in this discussion, and then send the link to Camille Avestruz via slack?

**Reply by evavagiakis** [reply]
I have written up an explanation of this plot and shared it to slack tagging Camille for discussion
